src/pfe/file-watcher/server/test/README.md

```diff
@@ -0,0 +1,104 @@
+# Codewind Turbine Tests
+
+## Test Suites
+
+The Codewind Turbine Module currently has two test suites at the moment:
+1. **Unit Test Suite**: This test suite calls each function directly without creating an instance of the turbine module. This is to test that we are able to write modular functions that can be tested standalone. This test suite also runs as part of the PR builds in the Codewind repo.
+
+2. **Functional Test Suite**: This test suite creates an instance of turbine as a node module and calls each actionable function (i.e Create, Delete, Update, etc.). This is the bigger test suite out of the two and can be ran on either a local Codewind instance or a hosted Che instance on kube.
+
+## How to run the test
+
+### Unit Test Suite
+
+Unit test suites are ran only on local. To run unit test on local, run the following commands:
+- `cd src/pfe/file-watcher/server/`
+- `npm i --only=dev`
+- `npm run unit:test`
+
+### Functional Test Suite
+
+Functional test suites can be either ran on your local Codewind instance or on a hosted Che instance on kube.
+
+#### Local
+
+To run the functional test on local, run the following commands:
+- `cd src/pfe/file-watcher/server/test/`
+- `./test.sh`
+
+#### Che (OpenShift Only)
+
+Running on kube requires some additional configurations such as passing the cluster ip, password and namespace.
+
+To run the functional test on kube, run the following commands:
+- `cd src/pfe/file-watcher/server/test/`
+- `CLUSTER_IP=<cluster_ip> CLUSTER_PORT=<cluster_port> CLUSTER_USER=<cluster_user> CLUSTER_PASSWORD=<cluster_password> ./test.sh -t kube -c y -p y`
+
+The options `-t` takes in the type of test environment to run, by default it is `local`. The options `-c` and `-p` stands for clean up before running and post clean up after test has finished.
+
+Additional options can be passed such as `IMAGE_PUSH_REGISTRY_ADDRESS`, `IMAGE_PUSH_REGISTRY_NAMESPACE` and `USER_DEVFILE` to overwrite the default values. 
+
+The default values are:
+- **IMAGE_PUSH_REGISTRY_ADDRESS**: docker-registry.default.svc:5000/che
+- **IMAGE_PUSH_REGISTRY_NAMESPACE**: eclipse-che
+- **USER_DEVFILE**: https://raw.githubusercontent.com/eclipse/codewind-che-plugin/master/devfiles/latest/devfile.yaml
+
+Don't have a Che instance? No worries, use the `che-setup.sh` script from https://github.com/eclipse/codewind-che-plugin/tree/master/scripts to deploy a che instance with codewind. More on how to deploy che automatically on the link above.
```

Comment:
Can probably just say `at the link` instead of `on the link above.` since it could be on the same line.

Reply:
Done